internal/clients/http/sources/sources_client_test.go

```diff
@@ -74,3 +74,71 @@ func TestSourcesClient_GetAuthentication(t *testing.T) {
 		assert.Equal(t, "arn:aws:iam::123456789999:role/redhat-provisioning-role-2f6d01c", authentication.Payload)
 	})
 }
+
+func TestSourcesClient_ListAllProvisioningSources(t *testing.T) {
+	mux := http.NewServeMux()
+	mux.HandleFunc(fmt.Sprintf("/application_types"), func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		_, err := io.WriteString(w, `{"data":[{"id":"5","created_at":"2023-02-02T16:05:56Z","updated_at":"2023-02-02T16:05:56Z","name":"/insights/platform/provisioning","display_name":"Provisioning","dependent_applications":[],"supported_source_types":["amazon","azure","google"],"supported_authentication_types":{"azure":["provisioning_lighthouse_subscription_id"],"amazon":["provisioning-arn"],"google":["provisioning_project_id"]}}],"meta":{"count":5,"limit":100,"offset":0},"links":{"first":"/api/sources/v3.1/application_types?limit=100&offset=0","last":"/api/sources/v3.1/application_types?limit=100&offset=100"}}`)
+		require.NoError(t, err, "failed to write http body for stubbed server")
+	})
+
+	mux.HandleFunc(fmt.Sprintf("/application_types/%s/sources", "5"), func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		_, err := io.WriteString(w, `{"data":[{"availability_status":"in_progress","id":"1","created_at":"2023-02-02T16:06:00Z","updated_at":"2023-02-02T16:06:00Z","name":"Amazon source","uid":"384ccb67-5838-4571-811f-d725a0a57b6f","app_creation_workflow":"manual_configuration","source_type_id":"1"},{"availability_status":"in_progress","id":"3","created_at":"2023-02-02T16:06:00Z","updated_at":"2023-02-02T16:06:00Z","name":"Google source","uid":"cb6fa946-395a-4650-ae6c-83501775ca4e","app_creation_workflow":"manual_configuration","source_type_id":"6"}],"meta":{"count":3,"limit":100,"offset":0},"links":{"first":"/api/sources/v3.1/sources?limit=100&offset=0","last":"/api/sources/v3.1/sources?limit=100&offset=100"}}`)
```

Comment:
Assuming this is copy pasted from real database, in the example above the "links":"first" path matches the handler path, but here I see a difference:

* handler path is `/application_types/%s/sources`
* and links is `/api/sources/v3.1/sources?limit=100&offset=0`

Just checking if this is just some typo in the test path or you pasted a wrong line.

Reply:
Yes a wrong copy-pasting, fixing